pkg/minishift/openshift/service.go

```diff
@@ -192,46 +205,65 @@ func calculateWeight(a float64, b float64) string {
 	return ""
 }
 
+func hasNodePort(serviceNodePort map[string]string, name string) bool {
+	_, ok := serviceNodePort[name]
+	return ok
+}
+
+func createServiceSpec(service ServiceWeight, routeURL, namespace, nodePort string) *ServiceSpec {
+	serviceSpec := &ServiceSpec{Name: service.Name,
+		Namespace: namespace,
+		Weight:    []string{service.Weight},
+		NodePort:  nodePort,
+	}
+
+	serviceSpec.URL = append(serviceSpec.URL, genUrl(service, routeURL))
+	return serviceSpec
+}
+
+func genUrl(service ServiceWeight, url string) string {
+	if service.Tls != "" {
+		return fmt.Sprintf(https, url)
+	}
+	return fmt.Sprintf(http, url)
+}
+
+func NewServiceSpecMap() *ServiceSpecMap {
+	return &ServiceSpecMap{S: make(map[string]*ServiceSpec)}
+}
+
+func (s *ServiceSpecMap) Add(service ServiceWeight, routeURL, namespace, nodePort string) {
+	if s.HasService(service.Name) {
+		s.S[service.Name].URL = append(s.S[service.Name].URL, genUrl(service, routeURL))
```

Comment:
See how it is being used here.   What is `s.S` means here?

Reply:
simple solution... but I hate to see 'implementation' details in naming (things like a ServiceString, EnableBool) and I see the same here. Maybe rather name it as follows:

`ServiceSpecMap` => `ServiceSpec`
and then you can use `S` => `Map`

note: and this fits exactly 3 characters.... how nice :-s. but still contains implementation details.

Reply:
I did not name S as something else, because we never directly interact with him..(in outside funtions) and s.S means S inside s.. thats what the `.` means.. :)

We always interact with `s *ServiceSpecMap` which has the functions `Add` and `hasService` for doing this.

I agree that the name of this needs to change..

Reply:
Just arbitrary rules about naming infuriate me (not that bad, just rather see a suggestion instead of merely criticism)